sdk/search/Azure.Search.Documents/tests/SearchServiceClientTests.cs

```diff
@@ -239,33 +234,70 @@ public async Task CreateAzureBlobIndexer()
-                {
-                    ClientRequestId = Recording.Random.NewGuid(),
-                });
+                GetOptions());
+
+            await WaitForIndexingAsync(serviceClient, actualIndexer.Name);
 
             // Run the indexer.
             await serviceClient.RunIndexerAsync(
                 indexer.Name,
-                new SearchRequestOptions
-                {
-                    ClientRequestId = Recording.Random.NewGuid(),
-                });
+                GetOptions());
 
-            // Indexers may take longer than indexing documents uploaded to the Search service.
-            await DelayAsync(TimeSpan.FromSeconds(5));
+            await WaitForIndexingAsync(serviceClient, actualIndexer.Name);
 
             // Query the index.
             SearchIndexClient indexClient = serviceClient.GetSearchIndexClient(
                 resources.IndexName);
 
             long count = await indexClient.GetDocumentCountAsync(
-                new SearchRequestOptions
-                {
-                    ClientRequestId = Recording.Random.NewGuid(),
-                });
+                GetOptions());
 
             Assert.AreEqual(SearchResources.TestDocuments.Length, count);
         }
+
+        /// <summary>
+        /// Gets a new <see cref="SearchRequestOptions"/>.
+        /// </summary>
+        /// <returns>
+        /// A new <see cref="SearchRequestOptions"/> with a new <see cref="SearchRequestOptions.ClientRequestId"/>.
+        /// </returns>
+        private SearchRequestOptions GetOptions() => new SearchRequestOptions
+        {
+            ClientRequestId = Recording.Random.NewGuid(),
+        };
+
+        /// <summary>
+        /// Waits for an indexer to complete up to the given <paramref name="timeout"/>.
+        /// </summary>
+        /// <param name="client">The <see cref="SearchServiceClient"/> to use for requests.</param>
+        /// <param name="indexerName">The name of the <see cref="SearchIndexer"/> to check.</param>
+        /// <param name="timeout">The amount of time before being canceled. The default is 1 minute.</param>
+        /// <returns>A <see cref="Task"/> to await.</returns>
+        private async Task WaitForIndexingAsync(
+            SearchServiceClient client,
+            string indexerName,
+            TimeSpan? timeout = null)
+        {
+            TimeSpan delay = TimeSpan.FromSeconds(10);
+            timeout ??= TimeSpan.FromMinutes(1);
+
+            using CancellationTokenSource cts = new CancellationTokenSource(timeout.Value);
+
+            while (true)
+            {
+                await DelayAsync(delay);
+
+                IndexerExecutionInfo status = await client.GetIndexerStatusAsync(
+                    indexerName,
+                    GetOptions(),
+                    cts.Token);
+
+                if (status.Status == IndexerStatus.Running &&
+                    status.LastResult?.Status == IndexerExecutionStatus.Success)
```

Comment:
The status stays `Running` after it's finished?

Reply:
That just means the indexer is running. I found it confusing at first as well. The docs aren't clear (I'm going to address this in our Teams channel) and it took samples to confirm my suspicions after seeing only Unknown (0), Running, and Error values.